alcs-frontend/src/app/features/admin/configuration/configuration.component.ts

```diff
@@ -3,37 +3,92 @@ import {
   AdminConfigurationService,
   CONFIG_VALUE,
 } from '../../../services/admin-configuration/admin-configuration.service';
+import { ConfirmationDialogService } from '../../../shared/confirmation-dialog/confirmation-dialog.service';
+import { MatSlideToggleChange } from '@angular/material/slide-toggle';
+import { MatDialog } from '@angular/material/dialog';
+import { MaintenanceBannerConfirmationDialogComponent } from './maintenance-banner-confirmation-dialog/maintenance-banner-confirmation-dialog.component';
+import { MaintenanceService } from '../../../services/maintenance/maintenance.service';
 
 @Component({
   selector: 'app-configuration',
   templateUrl: './configuration.component.html',
   styleUrls: ['./configuration.component.scss'],
 })
 export class ConfigurationComponent implements OnInit {
-  maintenanceMode = 'false';
-  isEditing = false;
+  maintenanceBanner = false;
+  maintenanceBannerMessage = '';
+  maintenanceMode = false;
 
-  constructor(private adminConfigurationService: AdminConfigurationService) {}
+  constructor(
+    private dialog: MatDialog,
+    private adminConfigurationService: AdminConfigurationService,
+    private maintenanceService: MaintenanceService,
+    protected confirmationDialogService: ConfirmationDialogService,
+  ) {}
 
   ngOnInit(): void {
     this.loadConfigs();
   }
 
-  toggleEdit() {
-    this.isEditing = !this.isEditing;
+  private async updateConfig(configName: CONFIG_VALUE, configValue: string) {
+    await this.adminConfigurationService.setConfiguration(configName, configValue);
+    this.loadConfigs();
   }
 
-  async onSave() {
-    await this.adminConfigurationService.setConfiguration(CONFIG_VALUE.PORTAL_MAINTENANCE_MODE, this.maintenanceMode);
-    this.loadConfigs();
-    this.isEditing = false;
+  private getConfigValue(
+    configs: { name: CONFIG_VALUE; value: string }[],
+    configName: CONFIG_VALUE,
+    isBoolean: boolean = false,
+  ): string | boolean {
```

Comment:
suggestion: Update this function to use generics so you don't need to do any casting on the calling side.

Reply:
Updated!